docs/utils/cookies.md

```diff
@@ -156,6 +156,48 @@ export async function loader({ request }: LoaderArgs) {
 }
 ```
 
+
+## Encrypting cookies
+
+It is possible to encrypt a cookie to automatically verify its contents when it is received. Since it's relatively easy to read even signed cookie, this is a good idea to carry sensitive data in cookie, like authentication information (see [sessions][sessions]).
+
+To encrypt a cookie, provide `encryption_key` when you first create the cookie:
+We use HKDF to derive encryption key so providing HASH is not recommended to avoid collisions best is to provide any text or hex you like can be random and we will handle key-size matching
+
+```js
+const cookie = createCookie("user-prefs", {
+  encryption_key: "my_encryption_secret",
```

Comment:
AES-GCM still has a useful lifetime, based on the number of encrypted messages. For example, [Google's docs](https://cloud.google.com/kms/docs/key-rotation).

`encryption_key` should be an array, so old cookies can be decrypted, but only signed with the newest secret.

Reply:
hey thanks for note added ability to rotate encryption cookie keys